app/src/main/res/layout/fragment_search_filter.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
```

Comment:
`ConstraintLayout` is to optimize and flatten the view hierarchy of your layouts by applying some rules to each view to avoid nesting.  Currently Only one filter is added , but in future other filter would also be added, so for then this would be better

Reply:
This layout can be easily achieved using linear layout and secondly it flatten the view heirarchy but at what cost and this is already single activity app and using constraint layout makes it even more expensive and time consuming so i would highly recommend using linear layout for such simple layouts 
References
https://android.jlelse.eu/constraint-layout-performance-870e5f238100

Reply:
And Linear layout slows down only if there is nesting level of >3 and in this case it wouldn't be happening ever

Reply:
Cool! thanks for the link. I would make changes surely.